src/Targeting/ActionHandler/ApplyTargetGroupsFromSegments.php

```diff
@@ -41,30 +37,19 @@ class ApplyTargetGroupsFromSegments implements ActionHandlerInterface, DataProvi
     const APPLY_TYPE_ONLY_MERGE = 'only_merge';
 
     /**
-     * @var SegmentManagerInterface
+     * @var DataLoaderInterface
      */
-    private $segmentManager;
-
-    /**
-     * @var SegmentTracker
-     */
-    private $segmentTracker;
+    protected $dataLoader;
 
     /**
      * @var TargetingStorageInterface
      */
     protected $storage;
 
     public function __construct(
-        ConditionMatcherInterface $conditionMatcher,
```

Comment:
I'd go for deprecation, instead of direct removal for BC reasons.

Reply:
Revert constructor changes and add TODO https://github.com/pimcore/customer-data-framework/pull/248/commits/83c5ac253ca423daaff63d02922b19d97b61fd01